src/pydvl/utils/parallel/map_reduce.py

```diff
@@ -70,8 +69,9 @@ def _(v: np.ndarray, *, timeout: Optional[float] = None) -> NDArray:
     return v
 
 
+# Careful to use list as hint. The dispatch does not work with typing generics
 @_get_value.register
-def _(v: Iterable, *, timeout: Optional[float] = None) -> List[Any]:
+def _(v: list, *, timeout: Optional[float] = None) -> List[Any]:
```

Comment:
This was a bug: dispatch needs the actual type, not generics (at least in our python versions)

Reply:
It did work for me. Which python version were you using?

The `Iterable` class from collections.abc is not the same as the one from typing.

Reply:
This breaks as soon as the remote function has string or some container which is not an NDArray as argument, and leads to an infinite recursion. I've hit the problem multiple times and IIRC it breaks with all 3 versions. I know it does with 3.8 on my laptop and 3.10 for sure.